src/scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/rest/MultiPortFilter.java

```diff
@@ -0,0 +1,54 @@
+package org.cloudfoundry.autoscaler.scheduler.rest;
+
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import lombok.extern.slf4j.Slf4j;
+import org.cloudfoundry.autoscaler.scheduler.conf.HealthServerConfiguration;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+@Order(1)
+public class MultiPortFilter implements Filter {
+  HealthServerConfiguration healthServerConfiguration;
+
+  public MultiPortFilter(HealthServerConfiguration healthServerConfiguration) {
+    this.healthServerConfiguration = healthServerConfiguration;
+  }
+
+  @Override
+  public void doFilter(
+      ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain)
+      throws IOException, ServletException {
+    HttpServletRequest httpRequest = (HttpServletRequest) servletRequest;
+    HttpServletResponse httpResponse = (HttpServletResponse) servletResponse;
+
+    // FIXME : refactor this duplicate block
+    if (isMainRequest(httpRequest)) {
+      log.debug("Main server request received on port " + healthServerConfiguration.getPort());
+      filterChain.doFilter(servletRequest, servletResponse);
+    } else if (isHealthRequest(httpRequest)) {
+      log.debug("Main server request received on port " + healthServerConfiguration.getPort());
+      filterChain.doFilter(servletRequest, servletResponse);
```

Comment:
If this is intentional to behave in exactly the same way, including writing in both cases about a "Main server request", I propose the following change:
```suggestion
    if (isMainRequest(httpRequest) || isHealthRequest(httpRequest)) {
      log.debug("Main server request received on port " + healthServerConfiguration.getPort());
      filterChain.doFilter(servletRequest, servletResponse);
```

But to me it looks as we actually wanted to call `httpRequest.getLocalPort()` instead of calling `healthServerConfiguration.getPort()`.

Reply:
This block is marked as Fix me annotation.. as we wanted to come back again at later stage :)


"But to me it looks as we actually wanted to call httpRequest.getLocalPort() instead of calling healthServerConfiguration.getPort() - > corrected